Marlin/src/inc/Conditionals_post.h

```diff
@@ -1936,206 +1928,281 @@
   #endif
 #endif
 
-//
-// Endstops and bed probe
-//
-
-#define _HAS_STOP(A,N,M,C) ((A##_HOME_TO_##M || (C+0)) && PIN_EXISTS(A##N##_##M))
+/**
+ * Endstop and probe flags
+ * - Set USE_(AXIS)_(MIN|MAX) flags for each used endstop that has a pin, including those for DIAG0 state.
+ *   - Note: Dual X Carriage uses "X" and "X2" steppers, but X_MIN and X_MAX endstop states (i.e., not X2_MAX).
+ * - Set a HAS_(AXIS)_(MIN|MAX)_STATE flag for each endstop that has a state, including SPI Sensorless which don't use a pin.
+ * - Set a HAS_(AXIS)_STATE flag for each axis that has at least one state.
+ */
+#define _USE_STOP(A,N,M,C) ((A##_HOME_TO_##M || (C+0)) && PIN_EXISTS(A##N##_##M) && !A##_SPI_SENSORLESS)
 
-#if _HAS_STOP(X,,MIN,)
-  #define HAS_X_MIN 1
+#if _USE_STOP(X,,MIN,)
+  #define USE_X_MIN 1
+#endif
+#if _USE_STOP(X,,MAX,ENABLED(DUAL_X_CARRIAGE))
+  #define USE_X_MAX 1
+#endif
+#if USE_X_MIN || BOTH(X_SPI_SENSORLESS, X_HOME_TO_MIN)
+  #define HAS_X_MIN_STATE 1
 #endif
-#if _HAS_STOP(X,,MAX,ENABLED(DUAL_X_CARRIAGE))
-  #define HAS_X_MAX 1
+#if USE_X_MAX || BOTH(X_SPI_SENSORLESS, X_HOME_TO_MAX)
+  #define HAS_X_MAX_STATE 1
 #endif
-#if HAS_X_MIN || HAS_X_MAX
-  #define HAS_X_ENDSTOP 1
+#if HAS_X_MIN_STATE || HAS_X_MAX_STATE
+  #define HAS_X_STATE 1
 #endif
 
-#if _HAS_STOP(Y,,MIN,)
-  #define HAS_Y_MIN 1
-#elif _HAS_STOP(Y,,MAX,)
-  #define HAS_Y_MAX 1
+#if _USE_STOP(Y,,MIN,)
```

Comment:
And for AXIS Y too;
#if _USE_STOP(Y,,MIN,ENABLED(HAS_DELTA_SENSORLESS_PROBING)))

Reply:
Delta Sensorless Probing remaps the "X, Y, and Z" MAX states (actually ABC, of course) to a single state, and this is never actually represented as an "endstop" state. It is merely used in the probing action:
```cpp
  const bool probe_triggered = (
    #if HAS_DELTA_SENSORLESS_PROBING
      endstops.trigger_state() & (_BV(X_MAX) | _BV(Y_MAX) | _BV(Z_MAX))
    #else
      TEST(endstops.trigger_state(), Z_MIN_PROBE)
    #endif
  );
```

Reply:
Something has changed, because now the MAX states are not activated. I have verified that the MIN states are activated instead. And I only get SENSORLESS_PROBING to work again if I change the code like this:
```
  const bool probe_triggered = (
    #if HAS_DELTA_SENSORLESS_PROBING
      endstops.trigger_state() & (_BV(X_MIN) | _BV(Y_MIN) | _BV(Z_MIN))
    #else
      TEST(endstops.trigger_state(), Z_MIN_PROBE)
    #endif
  );
```

Reply:
You point out a bug in this refactor, then. That is not intended to change, and should still be using the MAX states because that's where a Delta assigns places its endstops. For a Delta, only `[XYZ]_HOME_TO_MAX` should be defined, thus only `USE_[XYZ]_MAX` should be getting set here; only the `[XYZ]_MAX_PIN` states should be getting monitored in `enstops.cpp`; and only the `[XYZ]_MAX` endstop flags should be getting set. In fact there should be an error thrown for `_BV(_X_MIN)` and the others. So, I'll have to keep plugging away until these are all consistently working as intended.

Reply:
Indeed, as hoped and expected… When I use the configurations you provided to build with your suggested changes with this PR I get the error…
```
Marlin/src/module/probe.cpp:639:39: error: 'X_MIN' was not declared in this scope
       endstops.trigger_state() & (_BV(X_MIN) | _BV(Y_MIN) | _BV(Z_MIN))
```
Is it possible you've been tweaking your configuration in some way? I haven't made (very) significant changes to this PR in the last 24 hours.